src/apps/components/InstallWithManifestFormButton/InstallWithManifestFormButton.tsx

```diff
@@ -0,0 +1,94 @@
+import { TextField } from "@material-ui/core";
+import { Button } from "@saleor/components/Button";
+import { makeStyles } from "@saleor/macaw-ui";
+import React, { useState } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+type AvailableStates = "initial" | "input-open";
+
+const useStyles = makeStyles(
+  theme => ({
+    installButton: {
+      marginLeft: theme.spacing(2),
+      height: 52,
+    },
+  }),
+  {
+    name: "InstallWithManifestFormButton",
+  },
+);
+
+interface Props {
+  onSubmitted(manifestUrl: string): void;
+}
+
+export const InstallWithManifestFormButton = ({ onSubmitted }: Props) => {
+  const styles = useStyles();
+  const intl = useIntl();
+
+  const [state, setState] = useState<AvailableStates>("initial");
+
+  const handleFormSubmit: React.FormEventHandler<HTMLFormElement> = e => {
+    e.preventDefault();
+
+    const form = new FormData(e.currentTarget);
+    const inputValue = form.get("manifest-url") as string;
+
+    try {
+      new URL(inputValue);
+
+      onSubmitted(inputValue);
+    } catch (e) {
+      console.error("Invalid URL from input. Should be validated by browser");
```

Comment:
I wanted to avoid custom validation because its a complex topic. I used built in navigation in the browser - `required` and `type="url"` should handle all cases and it should be impossible to override it. This error here is an extra layer that should never happen. Adding custom errors here require rewriting entire form to be "controlled" and stateful. 

Are you sure we need it?